### ISSUE TITLE
Support emotion as an alternative to styled-components?

### ISSUE THREAD

**Issue by elliottsj** [grommet/grommet#2641]
Would grommet maintainers be open to supporting [emotion](https://emotion.sh/) as an alternative to styled-components?

For example, [rebass supports emotion](https://github.com/rebassjs/rebass/tree/118ab94fe87229bbb251c5cae2940c19dfeb34c9#emotion) via [a Babel config which replaces `'styled-components'` imports with `'@emotion/styled'`](https://github.com/rebassjs/rebass/blob/118ab94fe87229bbb251c5cae2940c19dfeb34c9/babel.config.js#L7-L16).

Off the top of my head, these are the changes that would need to be made in grommet:
* Configure emotion-specific rules in grommet's `.babelrc` so `'styled-components'` imports are renamed to `'@emotion/styled'`, ensure `withTheme` is imported from `'emotion-theming'`, etc.
* Remove styled-components from peerDependencies. Optionally use a run-time check instead to ensure a compatible version of either styled-components or emotion is installed.
* Update documentation / getting started guides to include steps for setting up with emotion.

There are probably other changes & challenges to face; would love to know what the community & maintainers think of this.

**Comment by ShimiSun** [comment]
shimi [11:36 AM]
@alansouzati @oorestisime I’m curios to know if you have tried `emotion` yourself and can list the the advantages it offers comparing to `styled-components`? or any other motivation reasons to pursue this direction.

alansouzati [11:36 AM]
https://github.com/styled-components/styled-components/issues/2418
this issue answers that

oorestisime [11:36 AM]
never tried

alansouzati [11:36 AM]
I haven't extensively tried but it is almost the same as styled-components
I'm wanting to create my own version of styled that does not use context
which is slow to render
see this for more details
https://github.com/styled-components/styled-components/issues/2215

**Comment by ShimiSun** [comment]
oorestisime [7:51 AM]
wow emotion is catching up reaaaaaaaaally fast https://npmcharts.com/compare/styled-components,emotion?interval=30

**Comment by krainboltgreene** [comment]
Just FYI, it's https://npmcharts.com/compare/styled-components,emotion,@emotion/core?interval=30

**Comment by alansouzati** [comment]
Although I feel number of install is nice, I don't think it should be that important.

I've tried to [compare](https://www.youtube.com/watch?v=AqFqduEEw7A&t=19s) both emotion and styled-components, they are relatively similar in terms of performance.

emotion is in fact faster but I'm still trying to find examples where emotion significantly excels over styled-components.

Both of them don't support hooks yet, so I'm definitely keeping an eye for this. Since I've observed that in development mode the performance is affected if you have a complex react tree and hooks will help with that.

I've mentioned to the core Grommet team about emotion. I think it is something we should definitely consider but this affects a lot of people and we need to be really careful when doing changes like this.

**Comment by chrischen** [comment]
Bundle size is also much smaller: https://bundlephobia.com/result?p=@emotion/core@10.0.10 <https://bundlephobia.com/result?p=@emotion/core@10.0.10> vs https://bundlephobia.com/result?p=styled-components@4.2.0 <https://bundlephobia.com/result?p=styled-components@4.2.0>

**Comment by brodybits** [comment]
I hope to give this a try in the near future (no promise right now). I wonder if this should target a new major release to avoid the risk of potentially nasty compatibility issues?

**Comment by ShimiSun** [comment]
>I wonder if this should target a new major release to avoid the risk of potentially nasty compatibility issues?
It most likely will be targeted to a new major release.

**Comment by sofiageo** [comment]
Would this replace styled components support or support both? We have picked grommet because of styled components support and we are writing custom styles with it at the moment.

**Comment by oorestisime** [comment]
Nope, i am thinking that compatibility with styled-components should be there. Supporting emotion would be a plus.

**Comment by perich** [comment]
Hi! is `emotion` support currently on the development roadmap or still in the planning phase currently? Would love to see it supported in Grommet
